src/main/scala/com/gossiperl/client/Messaging.scala

```diff
@@ -0,0 +1,73 @@
+package com.gossiperl.client
+
+import java.util
+
+import akka.actor.{Props, ActorLogging, Actor}
+import akka.util.Timeout
+import com.gossiperl.client.actors.ActorRegistry
+import com.gossiperl.client.exceptions.GossiperlClientException
+import com.gossiperl.client.serialization.{Serializer, DeserializeResultForward, DeserializeResultError, DeserializeResultOK}
+import com.gossiperl.client.thrift._
+import com.gossiperl.client.transport.{UdpTransportProtocol, UdpTransport}
+import scala.collection.JavaConverters._
+
+import scala.util.{Failure, Success}
+
+class Messaging(val configuration: OverlayConfiguration) extends ActorRegistry with ActorLogging {
+
+  import scala.concurrent.duration._
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  context.actorOf(Props( new UdpTransport(configuration) ), name = s"${configuration.overlayName}-transport")
+
+  log.debug(s"Messaging for overlay ${configuration.overlayName} is running.")
+
+  implicit val timeout = Timeout(1 seconds)
+
+  def receive = {
+    case GossiperlProxyProtocol.ShutdownRequest(p) =>
+      log.debug("Received shutdown request.")
+      p.success(self)
+      context.stop( self )
+    case UdpTransportProtocol.IncomingData(result) =>
+      result match {
+        case _:DeserializeResultOK =>
+          result.asInstanceOf[DeserializeResultOK].getDigestType match {
+            case Serializer.DIGEST =>
+              digestAck( result.asInstanceOf[DeserializeResultOK].getDigest.asInstanceOf[ Digest ] )
+            case Serializer.DIGEST_ACK =>
+              !:(s"${configuration.overlayName}-client-state", FSMProtocol.AckReceived)
```

Comment:
could you explain me what the "!:" operator does exactly? Haven't seen it before, so interested in what it does :)